### ISSUE TITLE
Building with qt 6.5 fails

### ISSUE THREAD

**Issue by Artturin** [hluk/CopyQ#2323]
```
/build/source/src/scriptable/scriptableproxy.cpp: In member function 'QByteArray ScriptableProxy::callFunctionHelper(const QByteArray&)':
/build/source/src/scriptable/scriptableproxy.cpp:924:23: error: no match for 'operator=' (operand types are 'QGenericArgument' and 'QMetaMethodArgument')
  924 |             args[i] = Q_ARG(QVariant, value);
      |                       ^~~~~
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note: candidate: 'constexpr QGenericArgument& QGenericArgument::operator=(const QGenericArgument&)'
   82 | class Q_CORE_EXPORT QGenericArgument
      |                     ^~~~~~~~~~~~~~~~
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from 'QMetaMethodArgument' to 'const QGenericArgument&'
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note: candidate: 'constexpr QGenericArgument& QGenericArgument::operator=(QGenericArgument&&)'
/nix/store/q5fzqbgghqf8nr947h8rvn80smrlr4rg-qtbase-6.5.0-dev/include/QtCore/qobjectdefs.h:82:21: note:   no known conversion for argument 1 from 'QMetaMethodArgument' to 'QGenericArgument&&'
/build/source/src/scriptable/scriptableproxy.cpp:952:17: error: operands to '?:' have different types 'QGenericReturnArgument' and 'QMetaMethodReturnArgument'
  952 |                 ? QGenericReturnArgument(returnValue.typeName(), static_cast<void*>(returnValue.data()) )
      |                 ^
```

**Comment by hluk** [comment]
That looks like a Qt bug. Q_ARG should return `QGenericArgument`: https://doc.qt.io/qt-6.5/qmetaobject.html#Q_ARG

I cannot find docs about `QMetaMethodArgument`.

**Comment by hluk** [comment]
I have not been able to test it on Qt 6.5 but this could fix the problem: #2324
